### ISSUE TITLE
Export * and export default

### ISSUE THREAD

**Issue by jkimbo** [benmosher/eslint-plugin-import#328]
I get the error `Multiple default exports` when trying to export everything from another file including the default export e.g.

```
export * from './foo';
export { default as default } from './foo';
```

It looks like that this case should throw an error (as shown by this [test case](https://github.com/benmosher/eslint-plugin-import/blob/90fefa3deb0853b4ab286eb07179fba62f4275c6/tests/src/rules/export.js#L32-L35)) however babel compiles the code `export * from './foo';` to:

```
Object.keys(_foo).forEach(function (key) {
  if (key === "default") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _foo[key];
    }
  });
});
```
[Reference](https://babeljs.io/repl/#?evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&code=export%20*%20from%20'.%2Ffoo'%3B)

so the second export is necessary to ensure the default export also gets exported.

Any suggestions? I'm not sure whether it's actually babel which is at fault here or if the assumption in this plugin is wrong. Happy to contribute a PR if a solution is agreed on.

**Comment by benmosher** [comment]
Interesting. I think the spec dictates that imported/exported namespaces include the `default` name; it's not distinct from other exported names, it just has special extra syntax for exporting/importing.

So I would argue to start that Babel is doing it wrong (maybe in the interest of backwards-compatibility?).

**Comment by jfmengels** [comment]
@jkimbo Are you running ESLint on the transpiled code? If you do, that could explain the error, but I don't see why you would do that.

**Comment by benmosher** [comment]
@jfmengels: if `./foo` contains a default export, it should report `Multiple default exports` on both lines.

@jkimbo: http://www.ecma-international.org/ecma-262/6.0/#sec-getexportednames, 7.d.i. seems to be filtering `default` exports from `export * from ...` statements. As such, looks like this plugin is in error and Babel has it right.

**Comment by jkimbo** [comment]
Thanks for digging into it @benmosher ! I can attempt a PR to fix it (though not until tomorrow) if that would help?

**Comment by jfmengels** [comment]
@jkimbo Yes please, that'd be great :)

**Comment by benmosher** [comment]
You're welcome to take a stab at it, but I think it goes pretty deep into the exported namespace tracking so it may get dicey. You may want to fail fast.
